host/sgx/calls.c

```diff
@@ -898,6 +899,41 @@ oe_result_t oe_call_enclave_function(
         output_bytes_written);
 }
 
+oe_result_t oe_call_synchronous_switchless_enclave_function(
+    oe_enclave_t* enclave,
+    oe_switchless_synchronous_ecall* node)
+{
+    oe_result_t result = OE_OK;
+
+    /* Reject invalid parameters */
+    if (NULL == enclave)
+    {
+        OE_RAISE(OE_INVALID_PARAMETER);
+    }
+
+    if (NULL == node)
+    {
+        OE_RAISE(OE_INVALID_PARAMETER);
+    }
+
+    /* start the switchless worker thread if needed */
+    oe_switchless_manager_startup(enclave);
+
+    /* set the synchrononization flag */
+    node->lock = 1;
+    oe_lockless_queue_push_back(
+        &(enclave->switchless_manager.switchless.ecall_queue), &(node->_node));
+
+    /* wait for the flag to clear */
+    while (__atomic_test_and_set(&(node->lock), __ATOMIC_ACQUIRE))
```

Comment:
Given that we are only waiting for **node->lock** to become 0, **__atomic_load** seems more appropriate here.

Reply:
This is using the atomic flag operations instead of the atomic value operations.  On some hardware, the flags are lighter weight, and they are also guaranteed to be atomic on all platforms where atomic values are not (but likely are).  For parity sake, __atomic_test_and_set is used on this thread while __atomic_clear is used on the other thread.